### ISSUE TITLE
Visual Studio can not find the Microsoft.SqlServer.Dac.Extensions file/assembly

### ISSUE THREAD

**Issue by Mattnificent** [microsoft/DacFx#311]
- DacFx Version: Microsoft.SqlServer.DacFx 162.0.52
- .NET Framework Version: 4.7.2
- Environment (local platform and source/target platforms): Windows 10

**Steps to Reproduce:**

1. Install Visual Studio 2022 with SSDT 
2. Create new SSDT project
3. Add .NET Framework 4.7.2 , console project
4. Install Microsoft.SqlServer.DacFx 162.0.52 package to the .NET 4.7.2 project
5. Right click the SSDT project, click Schema Compare
6. Select ANY 2 databases
7. Save the file
8. Create a Schema Compare object in the .NET Framework 4.7.2 project with a path to the Schema Compare file that was just created in the previous step
9. Run the program; you will get the following error

Could not load file or assembly 'Microsoft.SqlServer.Dac.Extensions, Version=162.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

**Did this occur in prior versions?  If not - which version(s) did it work in?**

This error appears to be happening in older versions as well.  Using the deprecated version, DacFx64, I was getting this error:

Could not load file or assembly 'Microsoft.SqlServer.Dac.Extensions, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

There appears to be a problem with the Nuget package dependencies.  It is not installing Microsoft.SqlServer.Dac.Extensions, or it is in some mystery directory that the Visual Studio devenv process can not find.

![image](https://github.com/microsoft/DacFx/assets/4649715/403e71a1-6b57-4df5-81c6-81a88eb3d102)

(DacFx/SqlPackage/SSMS/Azure Data Studio)

**Comment by Mattnificent** [comment]
I downgraded all of my projects to **.NET Framework 4.6.2** from .NET Framework 4.7.2, and now I am getting a different error message:
```
Cannot open database "<databaseName>" requested by the login. The login failed. 
Login failed for user '<computerName>\<userName>'
```

**Comment by Mattnificent** [comment]
Here is the full connection string that I am using:
```
Data Source=.;Initial Catalog=<databaseName>;Integrated Security=True;Persist Security Info=False;Pooling=False;Multiple Active Result Sets=False;Encrypt=False;Trust Server Certificate=True;Application Intent=ReadOnly;Command Timeout=0
```
I have tried running the process in CMD, still no cookie.

**Comment by Mattnificent** [comment]
Is this because the SqlAuthenticationMethod enum only has Active Directory, and SQL password?  Was Integrated Security deprecated?  Anyone have an article or an announcement on this?  Everything on StackOverflow is 10 years old now.  Is any of this stuff going to even work in 10 more years, or am I wasting my time, and wasting my life?

**Comment by Mattnificent** [comment]
So, apparently this error:
```
Cannot open database "<databaseName>" requested by the login. The login failed. 
Login failed for user '<computerName>\<userName>'
```
Was because I had a typo in the database name.  What a horribly misleading error message.  I guess "security" through lies is more important to our slave masters than empowering us with truth.

**Comment by Mattnificent** [comment]
As for the original issue, I have worked around the fact that this project does not appear to be supported by .NET Framework 4.7.2 by downgrading all of my projects to 4.6.2.
The way I figured this out was by going to `%UserProfile%\.nuget\packages\microsoft.sqlserver.dacfx\162.0.52\lib` where I only saw folders for .NET 6.0, 6.2, and Standard 2.1 - nothing for .NET 4.7.2.
![image](https://github.com/microsoft/DacFx/assets/4649715/33801ada-d37d-413d-a3b5-f61086c28671)
I'll leave this open in case anyone from Microsoft wants to add a build target for .NET 4.7.2 on this project.

**Comment by llali** [comment]
@Mattnificent I'm trying to repo this issue but I need more info about it. Can you please share part of you C# code to do the compare? I assume you run the Schema Compare in SSDT and the use the scmp file in a C# code. If I'm wrong please let me know. Thanks

**Comment by Mattnificent** [comment]
That is correct.  Here is the gist of the code:
```
SqlConnectionStringBuilder SourceDatabase = ...
SqlConnectionStringBuilder DestinationDatabase = ...
var schemaCompare = new SchemaComparison(
    new SchemaCompareDatabaseEndpoint(SourceDatabase.ConnectionString),
    new SchemaCompareDatabaseEndpoint(DestinationDatabase.ConnectionString)
);
SchemaComparisonResult comparisonResult = schemaCompare.Compare();
```

**Comment by llali** [comment]
@Mattnificent in general there shouldn't be any problem with using target 4.7.2 when using DacFx. 4.6.2 dlls should just work. You can see in other packages like https://www.nuget.org/packages/Microsoft.Data.SqlClient/ that they also have target 4.6.2 and you can use it in applications targeting 4.7.2. Now your issue might be related to something else. I tried to repro it with the test app you provided but mine works fine. I just created a simple c# console app and targeted to 4.7.2 and added DacFx nuget. Used you sample code, and it runs just fine. Can you verify the DacFx Extension dlls is added your project is coming from the 162.0.52 Nuget? and is there anything in the assembly binding that might cause any loading issue?

**Comment by Mattnificent** [comment]
Now that I think of it, when I changed all of my projects to run as .NET 4.6.2, it required me to install the .NET 4.6.2 SDK, if I recall correctly.  But the issue was happening on 2 different computers, so I think the NuGet packages were not resolving correctly or fully for some reason - probably because I created the source project on 2-3 generations of VS ago.  I couldn't even find the dll that VS was referencing.
